cgo_static.go

```diff
@@ -1,4 +1,4 @@
-//go:build !(duckdb_use_lib || duckdb_from_source) && (darwin || (linux && amd64))
+//go:build !duckdb_use_lib && (darwin || (linux && (amd64 || arm64)))
```

Comment:
This condition was correct before and is now incorrect. Maybe also a copy and paste error?